drivers/hmis/app/models/hmis/hud/enrollment.rb

```diff
@@ -148,11 +148,10 @@ def set_hud_project_id_from_project_pk_unless_wip
     search_term.strip!
 
     alpha_numeric = /[[[:alnum:]]-]+/.match(search_term).try(:[], 0) == search_term
-    numeric = /[\d-]+/.match(search_term).try(:[], 0) == search_term
 
-    # If numeric, check if it's an Enrollment primary key
-    if numeric
-      matching_enrollments = where(id: search_term)
+    # If it's a possible PK, check if it's an Enrollment primary key
+    if possibly_pk?(search_term)
+      matching_enrollments = where(id: search_term.to_i)
```

Comment:
> I was also wondering, could you help me identify where in the UI we have a free text search for Enrollments that uses the Enrollment search term filter? I'd like to test this too.

I believe this is when you go to the Project Dashboard => Enrollments => Clients and search. (And maybe you can do it on households table too, I forget).

Reply:
Thanks! I was confused about this because I couldn't see the bug get reproduced, even before I made the change in `enrollment.rb`.

I just stepped through it in a debugger and realized that this line,
```
matching_enrollments = where(id: search_term.to_i)
```
doesn't throw an error when search_term is '73892738928'. I was surprised by that because I couldn't see any difference in the database definitions of the ID columns between the Enrollment and Project models. Am I missing something very obvious?

Reply:
@martha good point. I can't find the rails documentation for this, but it does look throwing in a `to_i` on the project search (`p_t[:id].eq(search_term.to_i)`) fixes the issue for project search as well (i.e. no error is thrown).

I strangely can't get a minimal repro working for this. The query that fails in the UI works in the console. But both fail with `.to_sql`.

```
irb(main):007:0> Hmis::Hud::Project.where(Hmis::Hud::Project.arel_table[:id].eq('73892738928'))
=> []
irb(main):008:0> Hmis::Hud::Project.where(Hmis::Hud::Project.arel_table[:id].eq('73892738928')).to_sql
/bundle/ruby/3.1.0/gems/activemodel-7.0.8.1/lib/active_model/type/integer.rb:52:in `ensure_in_range': 73892738928 is out of range for ActiveModel::Type::Integer with limit 4 bytes (ActiveModel::RangeError)
irb(main):009:0> Hmis::Hud::Project.where(Hmis::Hud::Project.arel_table[:id].eq('73892738928'.to_i)).to_sql
/bundle/ruby/3.1.0/gems/activemodel-7.0.8.1/lib/active_model/type/integer.rb:52:in `ensure_in_range': 73892738928 is out of range for ActiveModel::Type::Integer with limit 4 bytes (ActiveModel::RangeError)
```

Soo... not totally sure what's going on, but I think the approach of a `possibly_pk?` method is good/safest.

Reply:
Huh, yeah that is surprising. I have no idea why either. One interesting note is that in the rails console the first query returns `[]`, but when stepping through in a debugger, calling the same query will return an `ActiveRecord_Relation` object, which only throws an error when you try to call something on it (like `.count` or `.compact`).

I did also notice in the debugger that calling `.to_i` in the context of Project still results in this error. So I'm really not sure what is different about Enrollments.

Confusing :/ but anyway, I agree that it's best to consistently use the `possibly_pk` approach for all 3.